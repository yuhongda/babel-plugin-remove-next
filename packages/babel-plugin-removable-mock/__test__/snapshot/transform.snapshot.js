export const resultTransformSnapshot = `
import { readFileSync } from "fs";
import { round } from "mathjs";
export const makeRound = v => {
  round(1.22, 2);

  return round(v);
};`

export const resultJSXTransformSnapshot = `
import React, { useState } from 'react';
import { observer } from 'mobx-react-lite';

const SFC_Inner_MyComponent = observer((props) => {
  const [counter, setCounter] = useState(0);

  return <div style={props.styles.blue}></div>;
});`


