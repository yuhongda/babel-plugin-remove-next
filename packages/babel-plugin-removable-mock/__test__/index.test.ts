import { transformSync } from "@babel/core";
import { format } from "prettier";
import presetTypescript from "@babel/preset-typescript";
import babelPluginRemovableMock from "../src/index";
import { resultTransformSnapshot } from "./snapshot/transform.snapshot";

export function transform(code: string) {
	const result = transformSync(code, {
		retainLines: true,
		presets: [[presetTypescript, { allExtensions: true, isTSX: true }]],
		plugins: [babelPluginRemovableMock],
	});

	if (result?.code == null) {
		return code;
	}

	return format(result.code, {
		parser: "babel",
		printWidth: 120,
		tabWidth: 2,
		singleQuote: true,
		semi: true,
		bracketSameLine: true,
		useTabs: false,
		trailingComma: "none",
	});
}

const variableDeclarationCase = `
import { readFileSync } from "fs";
import { round } from "mathjs";
export const makeRound = v => {
  // babel-plugin-removable-mocks
  const mocks = {
    d1: '11',
    v1: '11'
  };
  round(1.22, 2);

  return round(v);
};
`;

const functionDeclarationCase = `
import { readFileSync } from "fs";
import { round } from "mathjs";
export const makeRound = v => {
  // babel-plugin-removable-mocks
  function mocksFunc() {
		round(1.22, 2);
	};
  round(1.22, 2);

  return round(v);
};
`;

describe("general", function () {
	const variableDeclarationTransform = transform(variableDeclarationCase);
	const functionDeclarationTransform = transform(functionDeclarationCase);

	it("VariableDeclaration transform", () => {
		expect(variableDeclarationTransform).toMatchSnapshot(
			resultTransformSnapshot
		);
	});

	it("FunctionDeclaration transform", () => {
		expect(functionDeclarationTransform).toMatchSnapshot(
			resultTransformSnapshot
		);
	});
});
