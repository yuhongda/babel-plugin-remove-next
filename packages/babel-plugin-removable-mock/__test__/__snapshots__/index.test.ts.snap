// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`general FunctionDeclaration transform: 
import { readFileSync } from "fs";
import { round } from "mathjs";
export const makeRound = v => {
  round(1.22, 2);

  return round(v);
}; 1`] = `
"import { round } from 'mathjs';
export const makeRound = (v) => {
  round(1.22, 2);

  return round(v);
};
"
`;

exports[`general VariableDeclaration transform: 
import { readFileSync } from "fs";
import { round } from "mathjs";
export const makeRound = v => {
  round(1.22, 2);

  return round(v);
}; 1`] = `
"import { round } from 'mathjs';
export const makeRound = (v) => {
  round(1.22, 2);

  return round(v);
};
"
`;

exports[`normal general transform: 
import { readFileSync } from "fs";
import { round } from "mathjs";
export const makeRound = v => {
  round(1.22, 2);

  return round(v);
}; 1`] = `
"import { round } from 'mathjs';
export const makeRound = (v) => {
  round(1.22, 2);

  return round(v);
};
"
`;
